torch/onnx/_internal/fx/fx_onnx_interpreter.py

```diff
@@ -421,21 +455,28 @@ def _export_fx_node_to_onnxscript(
                     f"\nFound unsupported input types on PyTorch Op {node.target} with "
                     f"ValueError: \n{value_error}.\n"
                 )
-                diagnostic = diagnostic_context.inflight_diagnostic()
+                diagnostic = self.diagnostic_context.inflight_diagnostic()
                 diagnostic.with_additional_message(
                     f"### Op level debug fails due to unsupported input types\n"
                     f"{diagnostics.decorator.format_exception_in_markdown(value_error)}"
                 )
                 diagnostic.level = diagnostics.levels.ERROR
             else:
                 op_validation.validate_op_between_ort_torch(
-                    diagnostic_context, node, symbolic_fn, torch_args, torch_kwargs
+                    self.diagnostic_context, node, symbolic_fn, torch_args, torch_kwargs
                 )
-        fx_name_to_onnxscript_value[node.name] = output
-    elif node.op == "output":
+        self.fx_name_to_onnxscript_value[node.name] = output
+
+    @diagnostics.diagnose_call(
+        diagnostics.rules.fx_node_to_onnx,
+        diagnostic_message_formatter=_fx_node_to_onnx_message_formatter,
+    )
+    def output(self, node: torch.fx.Node):
         if isinstance(node.args[0], torch.fx.Node):
-            onnx_tensor_or_tensor_tuple = fx_name_to_onnxscript_value[node.args[0].name]
-            onnxscript_graph.register_outputs(onnx_tensor_or_tensor_tuple)
+            onnx_tensor_or_tensor_tuple = self.fx_name_to_onnxscript_value[
```

Comment:
Please pass in `fx_name_to_onnxscript_value` explicitly or define it as a local variable. Let's avoid stateful interactions via `self.something` class attributes.

Reply:
The idea is that `fx_name_to_onnxscript_value` is a property of `aten -> onnx` translation, which is responsibility of `FxOnnxInterpreter`. It is not just hiding parameter, but storing behavior + state in a cohesive class.

`fx_name_to_onnxscript_value` has no value outside this context, so keeping it dangling around where it is not used makes no sense.

Reply:
In other words, the split of each FX node implementation into its own method "caused" the creation of state. A huge if-else "solves" this at the expense of long spaghetti code.

I will try to find a solution with a local variable, but it is not out of this world for classes to have state. Every project with more than 1000 line have state. We just have to watch for not over doing

Reply:
The current code provides a stateless implementation for `split of each FX node implementation into its own method`. Why do we change this basic design principle in a housekeeping PR? I have tried very hard making this `virtual interpreter` as stateless as possible. This change basically discards the efforts. You don't need to find a solution. I have tried several strategies including writing `fx.Interpreter`. None of them describes the logics more explicitly than the current one. Remember as long as you make a class and use `self`, the future changes will likely use `self`. Let's avoid this by not using `self` in the beginning.

Reply:
done